kubric/assets/klevr.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2020 The Kubric Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from kubric.assets.asset_source import AssetSource
+from kubric.core import DirectionalLight, RectAreaLight, PerspectiveCamera, PrincipledBSDFMaterial
+from kubric.color import Color
+
+
+class KLEVR(AssetSource):
+  objects_list = [
+      "LargeMetalCube",
+      "LargeMetalCylinder",
+      "LargeMetalSphere",
+      "LargeRubberCube",
+      "LargeRubberCylinder",
+      "LargeRubberSphere",
+      "MetalSpot",
+      "RubberSpot",
+      "SmallMetalCube",
+      "SmallMetalCylinder",
+      "SmallMetalSphere",
+      "SmallRubberCube",
+      "SmallRubberCylinder",
+      "SmallRubberSphere",
+  ]
+
+  def __init__(self, uri: str):
+    super().__init__(uri)
+    self.spawn_region = ((-3.5, -3.5, 0), (3.5, 3.5, 2))
+    self.global_illumination = (0.05, 0.05, 0.05)
+
+  def get_scene_geometry(self):
+    return self.create(asset_id='Floor', static=True, position=(0, 0, -0.2))
```

Comment:
Yes, it would create two floors. But in that regard it is no different from the case of lights.
Neither the floor nor the lights are added to the scene twice, unless you also call `simulator.add(...)` or `renderer.add(...)` twice.
The only difference between using `self.create(...)` and constructing the object directly using `FileBasedObject(...)` is that the former "downloads" and unzips the relevant file.